src/main/java/teammates/ui/controller/Action.java

```diff
@@ -224,12 +224,12 @@ private boolean isNotLegacyLink() {
     }
 
     private boolean doesUserNeedToLogin(UserType currentUser) {
-        boolean userNeedsGoogleAccountForPage =
+        boolean shouldUserNeedGoogleAccountForPage =
                 !Const.SystemParams.PAGES_ACCESSIBLE_WITHOUT_GOOGLE_LOGIN.contains(request.getRequestURI());
-        boolean userIsNotLoggedIn = currentUser == null;
-        boolean noRegkeyGiven = getRegkeyFromRequest() == null;
+        boolean isUserNotLoggedIn = currentUser == null;
```

Comment:
I though sometimes it is not worth to enforce the negation rule. In this case, only one branch in `if` statement is used. I though `if (isUserNotLoggedIn) {...}` is better than `if (!isUserLoggedIn) { ... }`